### PR TITLE
First draft at GOV.UK on AWS introduction

### PR DESCRIPTION
**Pull request by surminus**
This is meant to be a short introduction to the differences between running GOV.UK in AWS. It should link to more detailed documents which we still need to write, so there are several broken links.

I would welcome suggestions to add relevant things, remove things that people don't need to immediately know, reformatting, rewording, basically I welcome everything ^_^

I am using this to kick start the documentation epic from the top.

https://trello.com/c/KMJB4DuI/862-documentation

### CODE REVIEW

File: source/manual/govuk-in-aws.html.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+#### PostgreSQL and MySQL
+
+We are using [Amazon Relational Database Service (RDS)](https://aws.amazon.com/rds/) to host PostgreSQL and MySQL databases.
+
+To run Puppet against these databases, we have a new instance class: [db_admin](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/node/s_db_admin.pp)
+
+Both PostgreSQL and MySQL databases are managed through this instance. They are also used to take textual backups, which are then stored in an [Amazon S3](https://aws.amazon.com/s3/) bucket.
+
+Transition has it's own class for management: [transition_db_admin](https://github.com/alphagov/govuk-puppet/blob/master/modules/govuk/manifests/node/s_transition_db_admin.pp)
```
Comment (issyl0) [inline source/manual/govuk-in-aws.html.md]: its

**Comment by danielroseman** [comment]
This might betray my ignorance, but if the host names and IPs are dynamic it would be useful to know how DNS is managed so that Plek etc works, since we can't map things like publishing-api.publishing.service.gov.uk in /etc/hosts.

File: source/manual/govuk-in-aws.html.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+### Deploying infrastructure
+
+[Terraform](https://terraform.io) is used to manage our [Infrastructure as Code](https://en.wikipedia.org/wiki/Infrastructure_as_Code). This code is stored
+in [govuk-aws](https://github.com/alphagov/govuk-aws).
+
+See the [documentation]() to make and deploy changes to the infrastructure.
+
+### Automated application deployments
+
+If an EC2 instance is terminated, it will be automatically rebuilt by the ASG. If an instance runs deployable applications,
```
Comment (nickcolley) [inline source/manual/govuk-in-aws.html.md]: What is ASG?

Comment (surminus) [reply source/manual/govuk-in-aws.html.md]: Auto Scaling Group. It was referenced previously in the document, but I will specify.

File: source/manual/govuk-in-aws.html.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+puppetmaster-1.management
+```
+
+Hostnames are now automatically generated by DHCP, and refer to the IP address and [region](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html) that the instance belongs to:
+
+```
+ip-10-1-4-100.eu-west-1.compute.internal
+```
+
+Please see [the documentation]() about accessing the environment.
```
Comment (issyl0) [inline source/manual/govuk-in-aws.html.md]: This is in #619.

**Comment by surminus** [comment]
@danielroseman I have added a section about DNS and service resolution - please let me know if it makes any sense.

File: source/manual/govuk-in-aws.html.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ip-10-1-4-100.eu-west-1.compute.internal
+```
+
+Please see [the documentation](howto-ssh-to-machines-in-aws.html) about accessing the environment.
+
+#### Service resolution
+
+Traditionally resolving a service name to an IP would be handled by hardcoding names and IPs in `/etc/hosts`.
+
+To make use of the dynamic environment in AWS, we are using [Amazon Route 53](https://aws.amazon.com/route53/) to resolve service names to their appropriate ELB. Each node group (a set of instances within an autoscaling group) will resolve a main service name, along with any application service names that belong to that group. For example, the `calculators-fronted` node group, will resolve `calculators-frontend` as the service name:
```
Comment (danielroseman) [inline source/manual/govuk-in-aws.html.md]: typo: `calculators-fronted`

**Comment by danielroseman** [comment]
@surminus looks great apart from one minor typo.

On another point though, the use of "blue" in the domain names presumably implies some kind of blue-green deployment process: is it worth explaining that anywhere?

**Comment by surminus** [comment]
This is actually being removed slightly, and I will update the docs to reflect that.

**Comment by surminus** [comment]
I've clarified, and added a broken link to some further docs I intend to write.

**Pull request merged**